### PR TITLE
Added function if

### PR DESCRIPTION
**Pull request by Pachinkos**
Added extension function if.

Example:
coerce.toString(condition.if($flow.num>5 && 1==1, 9999, "low"))

### CODE REVIEW

File: samples/extensions/TIBCO/function/condition/descriptor.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "condition",
+  "type": "flogo:function",
+  "version": "0.2.0",
```
Comment (lixingwang) [inline samples/extensions/TIBCO/function/condition/descriptor.json]: Maybe 0.1.0 would be better?

File: samples/extensions/TIBCO/function/condition/descriptor.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      "name": "if",
+      "description": "return the second or the third parameter",
+      "example": "if true return second parameter, else return the third parameter",
+      "args": [
+        {
+          "name": "condition",
+          "type": "boolean"
+        },
+        {
+          "name": "any1",
```
Comment (lixingwang) [inline samples/extensions/TIBCO/function/condition/descriptor.json]: How about change the name to **trueArg** and **falseArg** which easy to understand first look.

File: samples/extensions/TIBCO/function/condition/if.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+type fnIf struct {
+}
+
+func (fnIf) Name() string {
+	return "if"
+}
+
+func (fnIf) Sig() (paramTypes []data.Type, isVariadic bool) {
+	return []data.Type{data.TypeInt, data.TypeString, data.TypeString}, false
```
Comment (lixingwang) [inline samples/extensions/TIBCO/function/condition/if.go]: it should be data.TypeBool, data.TypeAny and data.TypeAny

File: samples/extensions/TIBCO/function/condition/descriptor.json (hunk 1 above)
Comment (Pachinkos) [reply samples/extensions/TIBCO/function/condition/descriptor.json]: Sure

File: samples/extensions/TIBCO/function/condition/descriptor.json (hunk 2 above)
Comment (Pachinkos) [reply samples/extensions/TIBCO/function/condition/descriptor.json]: Much more intuitive

File: samples/extensions/TIBCO/function/condition/if.go (hunk 3 above)
Comment (Pachinkos) [reply samples/extensions/TIBCO/function/condition/if.go]: Ok

**Pull request merged**